pytorch_transformers/modeling_utils.py

```diff
@@ -327,6 +327,14 @@ def _tie_or_clone_weights(self, first_module, second_module):
         else:
             first_module.weight = second_module.weight
 
+        if hasattr(first_module, 'bias') and first_module.bias is not None:
+            first_module.bias.data = torch.nn.functional.pad(
```

Comment:
It's a nice and concise way to do it but I'm worried about two things here:
- `torch.nn.functional.pad` is not present before pytorch 1.2.0 (and I think we should aim to keep compatibility with +1.0.1 if possible for now).
- when we reduce the size of the embeddings (which is supported right now), this will break, I believe.

Reply:
You're absolutely correct, I'm looking into it.

Reply:
I believe `torch.nn.functional.pad` was actually introduced in torch way back, and is available in the [documentation of version 1.0.0](https://pytorch.org/docs/1.0.0/nn.html?highlight=functional%20pad#torch.nn.functional.pad). I've run this code with torch 1.0.0 installed successfully.

The pad function actually accepts negative indices, and will then remove the overflowing elements. In this scenario, it removes the last elements, similar to the `resize_token_embeddings` method.

Here's an example, running on torch 1.0.0:

```py
from pytorch_transformers import XLNetTokenizer, XLNetLMHeadModel

model: XLNetLMHeadModel = XLNetLMHeadModel.from_pretrained("xlnet-base-cased")
tok = XLNetTokenizer.from_pretrained("xlnet-base-cased")
print(model.lm_loss.bias.shape, model.lm_loss.weight.shape)
# torch.Size([32000]) torch.Size([32000, 768])

tok.add_tokens(["token"])
model.resize_token_embeddings(len(tok))
print(model.lm_loss.bias.shape, model.lm_loss.weight.shape)
# torch.Size([32001]) torch.Size([32001, 768])

model.resize_token_embeddings(len(tok) - 100)
print(model.lm_loss.bias.shape, model.lm_loss.weight.shape)
# torch.Size([31901]) torch.Size([31901, 768])
```

Reply:
Damned, my doc search capabilities look quite rusty! Ok all good then :)